src/Simulator/IncrementalRedstoneSimulator/DaylightSensorHandler.h

```diff
@@ -0,0 +1,64 @@
+
+#pragma once
+
+
+
+
+
+namespace DaylightSensorHandler
+{
+	inline unsigned char GetPowerLevel(const cChunk & Chunk, const Vector3i Position)
+	{
+		int TimeOfDay = Chunk.GetWorld()->GetTimeOfDay();
+		bool IsInverted = (Chunk.GetBlock(Position) == E_BLOCK_DAYLIGHT_SENSOR) ? 0 : 1;
+		int Output = Clamp(FloorC(14 * std::sin((TimeOfDay * M_PI) / 12000) + 8), 0, 15);
+
+		if (IsInverted)
+		{
+			Output = 15 - Output;
+		}
+
+		return static_cast<unsigned char>(Output);
+	}
+
+	inline PowerLevel GetPowerDeliveredToPosition(const cChunk & a_Chunk, Vector3i a_Position, BLOCKTYPE a_BlockType, Vector3i a_QueryPosition, BLOCKTYPE a_QueryBlockType, bool IsLinked)
+	{
+		UNUSED(a_Chunk);
+		UNUSED(a_BlockType);
+		UNUSED(a_QueryPosition);
+
+		// Daylight sensors output to all surrounding blocks
+		// Retrieve and return the cached power calculated by Update for performance
+		return IsLinked ? 0 : DataForChunk(a_Chunk).GetCachedPowerData(a_Position);
```

Comment:
Sensors store their power level in the block metadata https://minecraft.gamepedia.com/Java_Edition_data_value/Pre-flattening#Daylight_Sensor so you can SetMeta/GetMeta directly.